File: src/main/java/com/example/CargoTracking/dto/InternationalAirReportPerformance.java
package com.example.CargoTracking.dto;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class InternationalAirReportPerformance {
    private Long id;
    private String preAlertNumber;
    private String referenceNumber;
    private String origin;
    private String destination;
    private String route;
    private String flight; // flight number
    private LocalDateTime actualTimeDeparture;
    private LocalDateTime actualTimeArrival;
    private LocalDateTime cleared;
    private Long totalTransitTime;
    private Long totalLeadTime;
}

File: frontend/src/app/model/InternationalAirReportPerformance.ts
export interface InternationalAirReportPerformance{
    id:number;
    preAlertNumber:string| null | undefined ;
    referenceNumber: string| null | undefined ;
    origin: string| null | undefined ;
    destination: string| null | undefined ;
    route:string| null | undefined ;
    flight:string | null | undefined;
    actualTimeDeparture :Date | string | null | undefined ;
    actualTimeArrival :Date | string | null | undefined ;
    cleared :Date | string | null | undefined ;
    totalTransitTime:number| null | undefined;
    totalLeadTime:number| null | undefined;
}